modules/config/keyboard.nix

```diff
@@ -0,0 +1,103 @@
+{ mode, config, lib, ... }:
+
+with lib;
+let
+  cfg = config.soxin.settings.keyboard;
+
+  layoutModule = types.submodule ({ config, ... }: {
+    options = {
+      layout = mkOption {
+        type = types.str;
+        default = "us";
+        example = "fr";
+        description = "Keyboard layout";
+      };
+
+      variant = mkOption {
+        type = types.str;
+        default = "";
+        example = "colemak";
+        description = "Keyboard variant";
+      };
+
+      keyMap = mkOption {
+        type = types.str;
+        example = "us";
+        description = ''
+          The keyboard mapping table for the virtual consoles. The default is
+          the layout associated with the keyMap but you may need to change this
+          depending on your keyboard variant.
+        '';
```

Comment:
Proposal: Let's do a section separation here between x11 settings and console settings. Also what section would fit best Wayland?

Reply:
I'm not sure about wayland. I guess that's a question for later, as we add graphical stuff.